packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -634,9 +691,20 @@ abstract class DiagnosticsNode {
   /// `:` is typically used as a separator when displaying as text.
   final bool showSeparator;
 
-  /// Whether the diagnostics should be hidden when showing the default
-  /// view of a tree.
-  bool get hidden;
+  /// Whether the diagnostic should be filtered due to its [level] being lower
+  /// than `minLevel`.
+  ///
+  /// If `minLevel` is [DiagnosticLevel.hidden] no diagnostics will be filtered.
+  /// If `minLevel` is [DiagnosticsLevel.off] all diagnostics will be filtered.
+  bool isFiltered(DiagnosticLevel minLevel) => level.index < minLevel.index;
+
+  /// Priority level of the diagnostic used to control which diagnostics should
+  /// be shown and filtered.
+  ///
+  /// Typically this only makes sense to set to a different value than
+  /// [DiagnosticLevel.info] for diagnostics representing properties where some
+  /// properties may be only marginally relevant.
+  DiagnosticLevel get level => DiagnosticLevel.info;
```

Comment:
either:
* this should probably mention how some subclasses have a "level" argument to their constructor which influences the value returned here, but that other factors also influence it,
or:
* the subclasses should provide their own overriding docs for this member when they change its behaviour

Reply:
Good idea. Done.